kiwiproject/kiwi
# Add @FieldRange validation annotation and validator

* Add @FieldRange validation annotation
* Add @FieldRanges annotation, which can hold multiple @FieldRange
  annotations
* Add overloaded KiwiValidations#addError method that accepts the
  property path the constraint violation should be attached to
* Add FieldRangeValidator, the validator class for @FieldRange
* Refactor common code shared by RangeValidator and FieldRangeValidator
  into package-private InternalKiwiValidators
* Update ValidationMessages.properties with the custom error message
  for violations of the @FieldRange annotation

Misc cleanup:

* Update InEnumValidator to use KiwiValidations.addError
* Update RangeValidator to use KiwiValidations.addError
* Update @Required#message() to use TEMPLATE_REQUIRED constant from
  new InternalKiwiValidators utility class

Fixes #309
Fixes #310
